xmodule/modulestore/mixed.py

```diff
@@ -828,34 +837,44 @@ def update_item(self, xblock, user_id, allow_not_found=False, **kwargs):  # lint
         Update the xblock persisted to be the same as the given for all types of fields
         (content, children, and metadata) attribute the change to the given user.
         """
-        store = self._verify_modulestore_support(xblock.location.course_key, 'update_item')
+        course_key = xblock.location.course_key
+        store = self._verify_modulestore_support(course_key, 'update_item')
         xblock = store.update_item(xblock, user_id, allow_not_found, **kwargs)
-        # .. event_implemented_name: XBLOCK_UPDATED
-        XBLOCK_UPDATED.send_event(
-            time=datetime.now(timezone.utc),
-            xblock_info=XBlockData(
-                usage_key=xblock.location.for_branch(None),
-                block_type=xblock.location.block_type,
-                version=xblock.location
+
+        def send_updated_event():
+            # .. event_implemented_name: XBLOCK_UPDATED
+            XBLOCK_UPDATED.send_event(
+                time=datetime.now(timezone.utc),
+                xblock_info=XBlockData(
+                    usage_key=xblock.location.for_branch(None),
+                    block_type=xblock.location.block_type,
+                    version=xblock.location
+                )
             )
-        )
+
+        store.on_commit_changes_to(course_key, send_updated_event)
         return xblock
 
     @strip_key
     def delete_item(self, location, user_id, **kwargs):  # lint-amnesty, pylint: disable=arguments-differ
         """
         Delete the given item from persistence. kwargs allow modulestore specific parameters.
         """
-        store = self._verify_modulestore_support(location.course_key, 'delete_item')
+        course_key = location.course_key
```

Comment:
@bradenmacdonald I had issues with deleting content from my course -- deleted blocks still show up in search?

But I'm not sure if there's something wrong with my setup, because I've got redis errors in the traceback.

```
cms-1  | 2024-05-22 23:47:11,093 ERROR 327 [django.dispatch] [user 6] [ip 172.18.0.1] dispatcher.py:213 - Error calling handle_item_deleted in Signal.send_robust() (BlockKey(type='vertical', id='f69a018879e5421f93fc631e8edd15da'))
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 211, in send_robust
cms-1  |     response = receiver(signal=self, sender=sender, **named)
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/signals/handlers.py", line 207, in handle_item_deleted
cms-1  |     deleted_block = modulestore().get_item(usage_key)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/mixed.py", line 90, in inner
cms-1  |     retval = func(field_decorator=strip_key_collection, *args, **kwargs)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/mixed.py", line 257, in get_item
cms-1  |     return store.get_item(usage_key, depth, **kwargs)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/split_draft.py", line 283, in get_item
cms-1  |     return super().get_item(usage_key, depth=depth, **kwargs)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/split.py", line 1183, in get_item
cms-1  |     items = self._load_items(course, [BlockKey.from_usage_key(usage_key)], depth, **kwargs)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/split.py", line 780, in _load_items
cms-1  |     return [runtime.load_item(block_key, course_entry, **kwargs) for block_key in block_keys]
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/split.py", line 780, in <listcomp>
cms-1  |     return [runtime.load_item(block_key, course_entry, **kwargs) for block_key in block_keys]
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/caching_descriptor_system.py", line 127, in _load_item
cms-1  |     block_data = self.get_module_data(block_key, course_key)
cms-1  |   File "/openedx/edx-platform/xmodule/modulestore/split_mongo/caching_descriptor_system.py", line 154, in get_module_data
cms-1  |     raise ItemNotFoundError(block_key)
cms-1  | xmodule.modulestore.exceptions.ItemNotFoundError: BlockKey(type='vertical', id='f69a018879e5421f93fc631e8edd15da')
cms-1  | 2024-05-22 23:47:11,094 INFO 327 [xmodule.modulestore.django] [user 6] [ip 172.18.0.1] django.py:220 - Sent item_deleted signal to <function handle_item_deleted at 0x7f831eca1e50> with kwargs {'usage_key': BlockUsageLocator(CourseLocator('SampleTaxonomyOrg2', 'STC1', '2023_1', 'draft-branch', None), 'vertical', 'f69a018879e5421f93fc631e8edd15da'), 'user_id': 6}. Response was: BlockKey(type='vertical', id='f69a018879e5421f93fc631e8edd15da')
cms-1  | 2024-05-22 23:47:11,200 INFO 327 [celery_utils.logged_task] [user 6] [ip 172.18.0.1] logged_task.py:25 - Task openedx.core.djangoapps.content.search.tasks.delete_xblock_index_doc[5f07654a-aeda-4b2d-b8f2-efd41ae30445] submitted with arguments ('block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@f69a018879e5421f93fc631e8edd15da',), {}
cms-1  | 2024-05-22 23:47:11,214 ERROR 327 [edx_event_bus_redis.internal.producer] [user 6] [ip 172.18.0.1] producer.py:51 - Error delivering message to Redis event bus. error=Error -2 connecting to edx.devstack.redis:6379. Name or service not known. full_topic='dev-course-authoring-xblock-lifecycle' event_key=None signal=<OpenEdxPublicSignal: org.openedx.content_authoring.xblock.deleted.v1> initial_topic='course-authoring-xblock-lifecycle' event_key_field='xblock_info.usage_key' event_data={'xblock_info': XBlockData(usage_key=BlockUsageLocator(CourseLocator('SampleTaxonomyOrg2', 'STC1', '2023_1', None, None), 'vertical', 'f69a018879e5421f93fc631e8edd15da'), block_type='vertical', version=None)} event_metadata=EventsMetadata(event_type='org.openedx.content_authoring.xblock.deleted.v1', id=UUID('9b992d54-1895-11ef-9c09-0242ac120009'), minorversion=0, source='openedx/cms/web', sourcehost='eab1bef4bb21', time=datetime.datetime(2024, 5, 22, 23, 47, 11, 147475, tzinfo=datetime.timezone.utc), sourcelib=(9, 9, 2))
```